roachpb/errors.go

```diff
@@ -172,6 +172,17 @@ func (e *Error) SetGoError(err error) {
 	}
 }
 
+// SetTxn sets the txn and resets the error message.
+func (e *Error) SetTxn(txn *Transaction) {
```

Comment:
Ah, I see. First tried to un-expose the field, but got the following error from `go vet`

```
roachpb/errors.pb.go:354: struct field txn has json tag but is not exported
```

Will rename to something `UnexposedTxn`..